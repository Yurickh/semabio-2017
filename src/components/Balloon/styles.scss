@import 'ambience/color';
@import 'ambience/responsive';
@import 'base/responsive-prop';

.balloon-bubble {
  display: flex;
  align-items: center;
  justify-content: center;

  box-sizing: border-box;
  width: 100%;

  @include responsive-prop(padding-top, 24, 24);
  @include responsive-prop(padding-bottom, 24, 24);
  @include responsive-prop(padding-left, 36, 36);
  @include responsive-prop(padding-right, 36, 36);

  background-color: #FFF;
  box-shadow: 0 2px 4px 0 rgba(119,119,119,.5);

  &.-question {
    font-weight: bold;
    @include responsive-prop(line-height, 17, 32);

    max-height: 80px;

    color: #58C166;
    border-radius: 100px;
    border-bottom-left-radius: 0;
  }

  &.-answer {
    @include responsive-prop(font-size, 18, 18);
    @include responsive-prop(line-height, 27, 27);
    @include responsive-prop(margin-top, 94, 94);

    color: $color-wall;
    border-radius: 40px;
    border-bottom-right-radius: 0;
  }
}

.example-enter {
  opacity: 0.01;
  width: 0px;
}

.example-enter.example-enter-active {
  opacity: 1;
  width: 100%;
  transition: 700ms;
}

.example-leave {
  opacity: 1;
  width: 100%;
}

.example-leave.example-leave-active {
  opacity: 0.01;
  width: 0px;
  transition: 700ms;
}
