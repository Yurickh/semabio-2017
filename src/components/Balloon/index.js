import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class Balloon extends Component {
	static propTypes = {
		type: PropTypes.oneOf(['question', 'answer']).isRequired,
		children: PropTypes.string.isRequired,
	}

	render() {
		return (
			<div className={`balloon-bubble -${this.props.type}`}>
				{this.props.children}
			</div>
		)
	}
}

export default Balloon
