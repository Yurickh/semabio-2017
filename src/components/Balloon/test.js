import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Balloon from '../Balloon'

describe('Balloon', () => {
  it('should render correctly', () => {
    const tree = shallow(<Balloon type="question">
      potato solid
    </Balloon>)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
