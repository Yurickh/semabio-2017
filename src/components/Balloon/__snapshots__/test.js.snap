// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balloon should render correctly 1`] = `
<div
  className="balloon-bubble -question"
>
  potato solid
</div>
`;
